lib/statem_dsl.ex

```diff
@@ -1,110 +1,105 @@
-  allowed, hence we return `true` without any further checking. This is also the
-  default implementation and the reason why the precondition is missing
-  in the test file.
+  and the generated list of arguments to check whether this call is allowed in
+  the current model state. In this particular example, `find` is always allowed,
+  hence we return `true` without any further checking. This is also the default
+  implementation and the reason why the precondition is missing in the test
+  file.
 
       defcommand :find do
         def impl(key), do: Cache.find(key)
-        def args(_state), do: [key()]
         def pre(_state, [_key]), do: true
       end
 
   If the precondition is satisfied, the call can happen. After the call, the SUT
   can be in a different state and the model state must be updated according to
-  the mapping of the SUT to the model. The function `next/3` takes the state before
-  the call, the list of arguments and the symbolic or dynamic result (depending
-  on phase 1 or 2, respectively). `next/3` returns the  new model state.  Since
-  searching for a key in the cache does not modify the system nor the model
-  state, nothing has to be done. This is again the default implementation and thus
-  left out in the test file.
+  the mapping of the SUT to the model. The function `next/3` takes the state
+  before the call, the list of arguments and the symbolic or dynamic result
+  (depending on phase 1 or 2, respectively). `next/3` returns the new model
+  state. Since searching for a key in the cache does not modify the system nor
+  the model state, nothing has to be done. This is again the default
+  implementation and thus left out in the test file.
 
       defcommand :find do
         def impl(key), do: Cache.find(key)
-        def args(_state), do: [key()]
         def pre(_state, [_key]), do: true
         def next(old_state, _args, call_result), do: old_state
       end
 
   The missing part of the command definition is the post condition, checking
-  that after calling the system in phase 2 the system is in the expected state
+  that after calling the system in phase 2, the system is in the expected state
   compared the model. This check is implemented in function `post/3`, which
-  again has a trivial default implementation for post conditions that always returns
-  true. In this example, we check if the `call_result` is `{:error, :not_found}`,
-  then we also do not find the key in our model list `entries`. The other case is
-  that if we a return value of `{:ok, val}`, then we also find the value via
-  the `key` in our list of `entries`.
+  again has a trivial default implementation for post conditions that always
+  returns true. In this example, we check if we can find the key in our list of
+  `entries` and if we do, we check if `call_result` resulted in `{:ok, val}`. Or
+  if we didn't found it, we check if SUT also couldn't find it by comparing if
```

Comment:
language: if we *don't find* it, we check if *the* SUT also *cannot* find

Reply:
✅